### DISCUSSION TITLE
Question regarding add_cdp_listener and clear_cdp_listeners

### DISCUSSION THREAD

**Discussion by oculunto** [seleniumbase/SeleniumBase#2758] [Q&A] [answered]
Hello, I'm using add_cdp_listener to capture and analyze some network data. 
 Sometimes it interests me to pause the callback, is there a way to do so ? Would adding an if / pass inside the callback be enough ? 
 Another question, this time about clear_cdp_listeners, does this method stops completely all the added cdp listeners added with add_cdp_listener or does it just clear the buffer (if there is any?).

**Comment by mdmintz** [comment, marked as answer]
Sounds like you're referring to this example: [SeleniumBase/examples/uc_cdp_events.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/uc_cdp_events.py).

Behind the scenes, that really just makes a call to `driver.get_log("performance")`, which comes from `driver.get_log()`, defined here: https://github.com/SeleniumHQ/selenium/blob/4b415384eae4ac156353726b473df35540625b88/py/selenium/webdriver/remote/webdriver.py#L1004

I don't really understand how it all works, but the original came from `undetected-chromedriver` - https://github.com/ultrafunkamsterdam/undetected-chromedriver/blob/0aa5fbe252370b4cb2b95526add445392cad27ba/undetected_chromedriver/__init__.py#L667

**Reply by oculunto** [reply]
Thank you, I will look into `driver.get_log()`
